cpp/oneapi/dal/algo/knn/backend/gpu/infer_kernel_brute_force_dpc.cpp

```diff
@@ -25,110 +23,251 @@
+
 #include "oneapi/dal/table/row_accessor.hpp"
 
+#include "oneapi/dal/detail/common.hpp"
+
 namespace oneapi::dal::knn::backend {
 
 using dal::backend::context_gpu;
 
 template <typename Task>
 using descriptor_t = detail::descriptor_base<Task>;
 
-namespace daal_classifier = daal::algorithms::classifier;
-namespace daal_knn = daal::algorithms::bf_knn_classification;
-namespace interop = dal::backend::interop;
+namespace bk = ::oneapi::dal::backend;
+namespace pr = ::oneapi::dal::backend::primitives;
 
 template <typename Float>
-using daal_knn_brute_force_kernel_t =
-    daal_knn::prediction::internal::KNNClassificationPredictKernelUCAPI<Float>;
+sycl::event sqrt(sycl::queue& q, array<Float>& data, const bk::event_vector& deps = {}) {
+    ONEDAL_ASSERT(data.has_mutable_data());
+    const auto length = data.get_count();
+    const auto range = bk::make_range_1d(length);
+    auto* const data_ptr = data.get_mutable_data();
+    return q.submit([&](sycl::handler& h) {
+        h.depends_on(deps);
+        h.parallel_for(range, [=](sycl::id<1> idx) {
+            data_ptr[idx] = sycl::sqrt(data_ptr[idx]);
+        });
+    });
+}
+
+template <typename Float>
+class knn_callback {
+    using voting_t = std::unique_ptr<pr::uniform_voting<std::int32_t>>;
+
+public:
+    knn_callback(sycl::queue& q,
+                 result_option_id results,
+                 std::int64_t query_block,
+                 std::int64_t query_length,
+                 std::int64_t k_neighbors)
+            : queue_(q),
+              result_options_(results),
+              query_block_(query_block),
+              query_length_(query_length),
+              k_neighbors_(k_neighbors),
+              temp_resp_(result_options_.test(result_options::responses)
+                             ? pr::ndarray<std::int32_t, 2>::empty(q,
+                                                                   { query_block, k_neighbors },
+                                                                   sycl::usm::alloc::device)
+                             : pr::ndarray<std::int32_t, 2>{}),
+              voting_(pr::make_uniform_voting(q, query_block, k_neighbors)) {}
+
+    auto& set_inp_responses(const array<std::int32_t>& inp_responses) {
+        this->inp_responses_ =
+            pr::ndarray<std::int32_t, 1>::wrap(inp_responses, inp_responses.get_count());
+        return *this;
+    }
+
+    auto& set_responses(array<std::int32_t>& responses) {
+        if (result_options_.test(result_options::responses)) {
+            ONEDAL_ASSERT(responses.get_count() == query_length_);
+            this->responses_ = pr::ndarray<std::int32_t, 1>::wrap_mutable(responses, query_length_);
+        }
+        return *this;
+    }
+
+    auto& set_indices(array<std::int32_t>& indices) {
+        if (result_options_.test(result_options::indices)) {
+            ONEDAL_ASSERT(indices.get_count() ==
+                          dal::detail::check_mul_overflow(query_length_, k_neighbors_));
```

Comment:
add namespace de = dal::detail; and use de:: to make it shorter

Reply:
Fixed